lib/internal/url.js

```diff
@@ -105,7 +105,11 @@ function parse(input, base) {
       this[context].query = query;
       this[context].fragment = fragment;
       this[context].host = host;
-      this[searchParams] = new URLSearchParams(query);
+      if (this[searchParams]) {  // invoked from href setter
+        initSearchParams(this[searchParams], query);
```

Comment:
So `initSearchParams` strips a leading question mark, if one exists. But is it possible for `query` contain a legitimate leading `?` at this point?

I'd rather just make a `parseSearchParams` function, while moving the stringification and question mark stripping to where those are needed.